src/kube-enricher.go

```diff
@@ -72,21 +75,31 @@ func main() {
 		informers.DebugInfo(log.Writer())
 	}
 
-	scanner := bufio.NewScanner(os.Stdin)
-	for scanner.Scan() {
-		in := scanner.Bytes()
-		enriched, err := enrich(informers, in, mapping)
+	var format format.Format = nil
+	switch *sourceFormat {
+	case "json":
+		format = jsonFormat.NewScanner(os.Stdin)
+	case "pb":
+		format = pbFormat.NewScanner(os.Stdin)
+	default:
+		log.Errorln("Unknown source format : ", sourceFormat)
```

Comment:
not tested yet but i think we should panic here (or alternatively use json as a fallback), to avoid subsequent panic on `format.Next` later